Sources/Algorithms/MinMax.swift

```diff
@@ -383,3 +383,141 @@ extension Collection where Element: Comparable {
+    while var low = iterator.next() {
+      if var high = iterator.next() {
+        // Update the upper bound with the larger new element.
+        if try areInIncreasingOrder(high, low) { swap(&low, &high) }
+        if try !areInIncreasingOrder(high, highest) { highest = high }
+      } else {
+        // Update the upper bound by reusing the last element.  The next element
+        // iteration will also fail, ending the loop.
+        if try !areInIncreasingOrder(low, highest) { highest = low }
+      }
+
+      // Update the lower bound with the smaller new element, which may need a
+      // swap first to determine.
+      if try areInIncreasingOrder(low, lowest) { lowest = low }
+    }
+    #else
+    /// Ensure the second argument has a value that is ranked at least as much as
+    /// the first argument.
+    func sort(_ a: inout Element, _ b: inout Element) throws {
+      if try areInIncreasingOrder(b, a) { swap(&a, &b) }
+    }
+
+    /// Find the smallest and largest values out of a group of four arguments.
+    func minAndMaxOf4(
+      _ a: Element, _ b: Element, _ c: Element, _ d: Element
+    ) throws -> (min: Element, max: Element) {
+      var (a, b, c, d) = (a, b, c, d)
+      try sort(&a, &b)
+      try sort(&c, &d)
+      try sort(&a, &c)
+      try sort(&b, &d)
+      return (a, d)
+    }
+
+    // Read the elements in four-at-a-time.  Some say this is more effective
+    // than a two-at-a-time loop.
+    while let a = iterator.next() {
+      let b = iterator.next() ?? a
+      let c = iterator.next() ?? b
+      let d = iterator.next() ?? c
+      let (low, high) = try minAndMaxOf4(a, b, c, d)
+      if try areInIncreasingOrder(low, lowest) { lowest = low }
+      if try !areInIncreasingOrder(high, highest) { highest = high }
+    }
+    #endif
+    return (lowest, highest)
+  }
+}
+
+extension Sequence where Element: Comparable {
+  /// Returns both the minimum and maximum elements in the sequence.
+  ///
+  /// This example finds the smallest and largest values in an array of height
+  /// measurements.
+  ///
+  ///     let heights = [67.5, 65.7, 64.3, 61.1, 58.5, 60.3, 64.9]
+  ///     if let (lowestHeight, greatestHeight) = heights.minAndMax() {
+  ///         print(lowestHeight, greatestHeight)
+  ///     } else {
+  ///         print("The list of heights is empty")
+  ///     }
+  ///     // Prints: "58.5 67.5"
+  ///
+  /// - Precondition: The sequence is finite.
+  ///
+  /// - Returns: A tuple with the sequence's minimum element, followed by its
+  ///   maximum element. For either member, if there is a tie for the extreme
+  ///   value, the element chosen is unspecified. The same element may be used
+  ///   for both members if all the elements are equal. If the sequence has no
+  ///   elements, returns `nil`.
+  ///
+  /// - Complexity: O(*n*), where *n* is the length of the sequence.
+  @inlinable
+  public func minAndMax() -> (min: Element, max: Element)? {
+    return minAndMax(by: <)
```

Comment:
Nit: We don't need a return here.

Reply:
I like the `return`.

As a matter of style, I think `return` should only be omitted when the entire`{...}` block is on a single line.

Reply:
I wish we had a style guide for these things.